source/dem/particle_particle_contact_force.cc

```diff
@@ -469,6 +479,518 @@ ParticleParticleContactForce<dim, contact_model>::
+                        this->normal_force,
+                        this->tangential_force,
+                        this->particle_one_tangential_torque,
+                        this->particle_two_tangential_torque,
+                        this->rolling_resistance_torque);
+                    }
+
+                  if constexpr (contact_model ==
+                                Parameters::Lagrangian::
+                                  ParticleParticleContactForceModel::
+                                    hertz_mindlin_limit_force)
+                    {
+                      calculate_hertz_mindlin_limit_force_contact(
+                        contact_info,
+                        this->normal_relative_velocity_value,
+                        this->normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        this->normal_force,
+                        this->tangential_force,
+                        this->particle_one_tangential_torque,
+                        this->particle_two_tangential_torque,
+                        this->rolling_resistance_torque);
+                    }
+
+                  if constexpr (contact_model ==
+                                Parameters::Lagrangian::
+                                  ParticleParticleContactForceModel::
+                                    hertz_mindlin_limit_overlap)
+                    {
+                      calculate_hertz_mindlin_limit_overlap_contact(
+                        contact_info,
+                        this->normal_relative_velocity_value,
+                        this->normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        this->normal_force,
+                        this->tangential_force,
+                        this->particle_one_tangential_torque,
+                        this->particle_two_tangential_torque,
+                        this->rolling_resistance_torque);
+                    }
+
+
+                  // Apply the calculated forces and torques on the particle
+                  // pair
+                  this->apply_force_and_torque_on_ghost_particles(
+                    this->normal_force,
+                    this->tangential_force,
+                    this->particle_one_tangential_torque,
+                    this->rolling_resistance_torque,
+                    particle_one_torque,
+                    particle_one_force);
+                }
+
+              else
+                {
+                  // if the adjacent pair is not in contact anymore, only the
+                  // tangential overlap is set to zero
+                  contact_info.tangential_overlap.clear();
+                }
+            }
+        }
+    }
+
+  for (auto &&periodic_adjacent_particles_list :
```

Comment:
Yeah ok clearly refactoring these loops will be a good thing to do in a next PR.